apps/system/lockscreen/js/widgets/clock/lockscreen_clock_widget.js

```diff
@@ -50,6 +52,56 @@
     this.logger.debug('Clock updated', now);
   };
 
+  LockScreenClockWidget.prototype.updateAlarm =
+	function() {
+    this.fetchAlarmData()
+    .then((alarmData) => {
+
+      // if there is no alarm
+      if(!alarmData.hour && !alarmData.minute) {
+        this.resources.elements.alarm.classList.add('no-alarms');
+        return;
+      }
+
+      var type = 'AM',
+          hour = parseInt(alarmData.hour, 10);
+
+      // Decide wheter to use AM or PM
+      if(hour > 12) {
+        type = 'PM';
+        hour = hour - 12;
+      }
+
+      var fullAlarmTime = hour + ':' + alarmData.minute + type;
+      this.resources.elements.alarmtime.textContent = fullAlarmTime;
+      this.resources.elements.alarm.classList.remove('no-alarms');
+      this.resources.elements.alarm.classList.add('has-alarm');
+      this.logger.debug('Alarm updated to :', fullAlarmTime);
+    }).catch(() => {
```

Comment:
And I suggest you that to add a `console.error` here (before you re-throw the error). To log the real error is a good practise for using `Promise`. Since if you neither do that nor re-throw the error, you'll **silent** the error. This is a real pain point during debugging `Promise`.

(That is, after you change the usage of `catch`).